Added libretro cores

Menelkir opened the pull request:
Hi, I've added a couple of things.  

- Added homura: https://github.com/the-homura-project/homura
- I've got the libretro-cores that was deprecated on official freebsd ports, updated to the last verified commit every core that I was able to compile and test on FreeBSD 12.0. Some ports needs extra work and it's beyond my freebsd knowledge. 

**Broken ports:** 
games/libretro-parallel_n64:    Needs proper GL fixes
games/libretro-mupen64:         Need some work
games/libretro-desmume:         Need some work, use desmume2015 for now
games/libretro-ppsspp:          Need external repos
games/libretro-citra:           Need external repos
games/libretro-flycast          Need some work

valpackett committed:
deskutils/gnome-maps: update

valpackett committed:
deskutils/gnome-maps: update

valpackett committed:
games/retroarch: 1.7.5

valpackett committed:
games/retroarch: 1.7.5

valpackett committed:
Make the used mesa port a variable

valpackett committed:
Make the used mesa port a variable

kwm81 committed:
Update vala to 0.44.5.

kwm81 committed:
Update vala to 0.44.5.

kwm81 committed:
Update gobject-introspection to 1.60.2.

kwm81 committed:
Update gobject-introspection to 1.60.2.

valpackett committed:
graphics/mesa-dev: new better mesa port

valpackett committed:
graphics/mesa-dev: new better mesa port

valpackett committed:
devel/evemu: upd

valpackett committed:
devel/evemu: upd

valpackett committed:
graphics/colord: vapi

valpackett committed:
graphics/colord: vapi

valpackett committed:
devel/mull

valpackett committed:
devel/mull

valpackett committed:
devel/libgusb

valpackett committed:
devel/libgusb

valpackett committed:
games/dose-response

valpackett committed:
games/dose-response

valpackett committed:
graphics/rawtherapee: mmap/lto

valpackett committed:
graphics/rawtherapee: mmap/lto

valpackett committed:
math/nasc

valpackett committed:
math/nasc

valpackett committed:
editors/abiword: upd

valpackett committed:
editors/abiword: upd

valpackett committed:
games/vcmi (xxx: server crash)

valpackett committed:
games/vcmi (xxx: server crash)

valpackett committed:
accessibility/redshift-wlr

valpackett committed:
accessibility/redshift-wlr

valpackett committed:
graphics/simple-scan [dep: libgusb]

valpackett committed:
graphics/simple-scan [dep: libgusb]

valpackett committed:
games/regoth

valpackett committed:
games/regoth

valpackett committed:
audio/sonata: upd (gtk3)

valpackett committed:
audio/sonata: upd (gtk3)

valpackett committed:
cad/solvespace: upd (gtk3)

valpackett committed:
cad/solvespace: upd (gtk3)

kwm81 committed:
Update dconf to 0.32.0.

kwm81 committed:
Update dconf to 0.32.0.

kwm81 committed:
Update dconf-editor to 3.32.0.

kwm81 committed:
Update dconf-editor to 3.32.0.

kwm81 committed:
Update gnome-calculator to 3.32.2.

kwm81 committed:
Update gnome-calculator to 3.32.2.

valpackett committed:
audio/jack: jack2

valpackett committed:
audio/jack: jack2

valpackett committed:
graphics/inkscape: 1.0alpha

valpackett committed:
graphics/inkscape: 1.0alpha

valpackett committed:
x11/kitty

valpackett committed:
x11/kitty

valpackett committed:
graphics/hdrmerge

valpackett committed:
graphics/hdrmerge

valpackett committed:
readme

valpackett committed:
readme

valpackett committed:
devel/glslang: upd

valpackett committed:
devel/glslang: upd

valpackett committed:
devel/libudev-devd: my version

valpackett committed:
devel/libudev-devd: my version

valpackett committed:
graphics/gifup

valpackett committed:
graphics/gifup

kwm81 committed:
Add spidermonkey 60.x

kwm81 committed:
Update gjs to 1.56.2.

kwm81 committed:
Update gjs to 1.56.2.

kwm81 committed:
Update nautilus to 3.32.1.

kwm81 committed:
Update nautilus to 3.32.1.

kwm81 committed:
Update evince to 3.32.0.

kwm81 committed:
Update evince to 3.32.0.

kwm81 committed:
Update pygobject3 to 3.32.2.

kwm81 committed:
Update pygobject3 to 3.32.2.

valpackett committed:
sysutils/consolekit2: wayland

valpackett committed:
sysutils/consolekit2: wayland

valpackett committed:
gnome-terminal: fix plist

valpackett committed:
gnome-terminal: fix plist

valpackett committed:
sysutils/grub2: partially fix this stuff

valpackett committed:
sysutils/grub2: partially fix this stuff

valpackett committed:
clutter, cogl, libchamlain: enable wayland

valpackett committed:
clutter, cogl, libchamlain: enable wayland

kwm81 committed:
Update glib-networking 2.60.3.

kwm81 committed:
Update glib-networking 2.60.3.

kwm81 committed:
Update jsonrpc-glib to 3.32.0.

kwm81 committed:
Update jsonrpc-glib to 3.32.0.

kwm81 committed:
Update gnome-calendar to 3.32.2.

kwm81 committed:
Update gnome-calendar to 3.32.2.

kwm81 committed:
Update gnome-weather to 3.32.2.

kwm81 committed:
Update gnome-weather to 3.32.2.

kwm81 committed:
Update libgtop to 2.40.0.

kwm81 committed:
Update libgtop to 2.40.0.

kwm81 committed:
Update gnome-system-monitor to 3.32.1.

kwm81 committed:
Update atkmm to 2.28.0.

kwm81 committed:
Update atkmm to 2.28.0.

kwm81 committed:
Update gedit to 3.32.2.

kwm81 committed:
Update gedit to 3.32.2.

kwm81 committed:
Update gedit-plugins to 3.32.2.

kwm81 committed:
Update gedit-plugins to 3.32.2.

valpackett committed:
multimedia/gstreamer1-plugins: backport fix for GL sink on Wayland compositors w/o wl-shell

This is ugly (in the makefile)

valpackett committed:
multimedia/gstreamer1-plugins: backport fix for GL sink on Wayland compositors w/o wl-shell

This is ugly (in the makefile)

valpackett committed:
graphics/mypaint: 2.0.0 alpha

valpackett committed:
graphics/mypaint: 2.0.0 alpha

kwm81 committed:
Update gucharmap to 11.0.3.

kwm81 committed:
Update gucharmap to 11.0.3.

kwm81 committed:
Update libgweather to 3.32.2.

kwm81 committed:
Update libgweather to 3.32.2.

kwm81 committed:
Update glib20 to 2.60.4.

kwm81 committed:
Update glib20 to 2.60.4.

valpackett committed:
devel/libunwind: upd

valpackett committed:
devel/libunwind: upd

kwm81 committed:
Update template-glib to 3.32.0.

kwm81 committed:
Update template-glib to 3.32.0.

kwm81 committed:
Update gtksourceview4 to 4.2.0.

kwm81 committed:
Update gtksourceview4 to 4.2.0.

kwm81 committed:
Update glibmm to 2.60.0.

kwm81 committed:
Update glibmm to 2.60.0.

kwm81 committed:
Update pangomm to 2.42.0.

kwm81 committed:
Update pangomm to 2.42.0.

kwm81 committed:
Misc cairomm cleanups

kwm81 committed:
Misc cairomm cleanups

kwm81 committed:
Add cairomm 1.15.5, API 1.16

kwm81 committed:
Add cairomm 1.15.5, API 1.16

kwm81 committed:
Update gtkmm30 to 3.24.1.

kwm81 committed:
Update gtkmm30 to 3.24.1.

kwm81 committed:
add vala to gnome-usage

kwm81 committed:
add vala to gnome-usage

valpackett committed:
multimedia/libde265: upd

valpackett committed:
multimedia/libde265: upd

valpackett committed:
Add dummies

valpackett committed:
Add dummies

valpackett committed:
New port: x11/tilix, x11-toolkits/gtkd-3

valpackett committed:
New port: x11/tilix, x11-toolkits/gtkd-3

valpackett committed:
lang/elixir: build with otp 22

valpackett committed:
lang/elixir: build with otp 22

valpackett committed:
mail/geary: 3.33.1 https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=238373

valpackett committed:
mail/geary: 3.33.1 https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=238373

valpackett committed:
print/latex-biber: update to 2.12 (for texlive 2019 official installer)

See also https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=227332
https://reviews.freebsd.org/D17836

valpackett committed:
print/latex-biber: update to 2.12 (for texlive 2019 official installer)

See also https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=227332
https://reviews.freebsd.org/D17836

valpackett committed:
graphics/rawtherapee: openmp idle full load fix

valpackett committed:
graphics/rawtherapee: openmp idle full load fix

valpackett committed:
Add deskutils/bookworm

valpackett committed:
Add deskutils/bookworm

valpackett committed:
Add audio/loudness-scanner

valpackett committed:
Add audio/loudness-scanner

valpackett committed:
Add multimedia/gnome-podcasts

valpackett committed:
Add multimedia/gnome-podcasts

valpackett committed:
Purge xorg-fonts deps

Thread: https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=238482

valpackett committed:
Purge xorg-fonts deps

Thread: https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=238482

valpackett committed:
graphics/graphene: 1.9.4

valpackett committed:
graphics/graphene: 1.9.4

valpackett committed:
gtk-common dummy for getting off ports-gnome

valpackett committed:
gtk-common dummy for getting off ports-gnome

valpackett committed:
x11-toolkits/gtk40: 3.96.0

valpackett committed:
x11-toolkits/gtk40: 3.96.0

kwm81 committed:
Update vte3 to 0.56.3.

kwm81 committed:
Update vte3 to 0.56.3.

valpackett committed:
devel/libudev-devd: just latest commit

valpackett committed:
devel/libudev-devd: just latest commit

valpackett committed:
devel/llvm80: OpenMP and sanitizers on aarch64

Sanitizer patches are sorta WIP, but ASAN, libFuzzer, UBSAN, CFI,
and even HWASAN work. Currently requires kernel patches mentioned
in the patch files here.

valpackett committed:
devel/llvm80: OpenMP and sanitizers on aarch64

Sanitizer patches are sorta WIP, but ASAN, libFuzzer, UBSAN, CFI,
and even HWASAN work. Currently requires kernel patches mentioned
in the patch files here.

valpackett committed:
sysutils/consolekit2: upd patches

valpackett committed:
sysutils/consolekit2: upd patches

valpackett committed:
lang/ldc: 1.16.0 / aarch64 https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=237427

valpackett committed:
lang/ldc: 1.18.0 / aarch64 https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=237427

valpackett committed:
games/supertuxkart: wayland + unBROKEN https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=239141

valpackett committed:
pygobj plist fix

valpackett committed:
pygobj plist fix

valpackett committed:
devel/ispc: git master - adds aarch64

valpackett committed:
devel/ispc: git master - adds aarch64

valpackett committed:
graphics/xournalpp: do not depend on latex

valpackett committed:
graphics/xournalpp: do not depend on latex

valpackett committed:
x11/libinput: update to 1.14.1 https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=240153

valpackett committed:
x11/libinput: update to 1.14.1 https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=240153

valpackett committed:
x11/pixman: import aarch64 neon patch

valpackett committed:
x11/pixman: import aarch64 neon patch

valpackett committed:
devel/libepoll-shim: upstream

valpackett committed:
devel/libepoll-shim: upstream

valpackett committed:
x11-fonts/paratype https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=240514

valpackett committed:
x11-toolkits/gtk30: update to 3.24.12

valpackett committed:
x11-toolkits/gtk30: update to 3.24.12

Commit:
Added libretro cores

myfreeweb commented:
Wow, this is cool!

- Upstream emulator ports that provide cores (mgba, ppsspp, mesen) currently use `lib/libretro` not `libexec/libretro` (which makes sense, shared libraries are not executables - idk why these things install into libexec by default?). This should be consistent.
- Looks like the same install procedure copy-pasted into all the core ports? This should be reused..
- You can use [portfmt](https://github.com/t6/portfmt) to fix formatting (e.g. I see `RUN_DEPENDS` on a single line in homura)
- You should submit new ports directly into the upstream [bugzilla](https://bugs.freebsd.org/) :)

/cc @yurivict

Menelkir commented:
I actually (mostly) copy/pasted. The homura I copied and tested, the original repo have a more updated one from time to time. The libretro ports I didn't changed too much of what used to be in the late libretro-cores port but I've tried to let them more up-to-date.

Commit:
modified:   games/libretro-4do/Makefile
	modified:   games/libretro-beetle_pce_fast/Makefile
	new file:   games/libretro-beetle_pce_fast/distinfo
	modified:   games/libretro-gambatte/Makefile
	modified:   games/libretro-genesis_plus_gx/Makefile
	modified:   games/libretro-snes9x/Makefile
	modified:   games/libretro-uae/Makefile

Menelkir commented:
I've missed a distinfo, fixed some licenses and marked uae as borked. Now everything should compile as expected (tested on 12.1).

valpackett committed:
devel/libclc: update

valpackett committed:
graphics/libdrm: 2.4.100

valpackett committed:
Set PULSEAUDIO and other defaults in a bunch of places

valpackett committed:
NEW PORT sysutils/hidviz

Commit:
Merge branch 'lite' of https://github.com/myfreeweb/freebsd-ports-dank into lite

Commit:
Fixing License issues
